added edit contact

lesleykhuu opened the pull request:
added edit contact. used helper functions in Contact.php

lesleykhuu committed:
added edit contact

added edit contact. used helper functions in Contact.php

lesleykhuu committed:
added GetById

added GetById and changed Get to GetByEmail. Added contactId column in
contactlist table

lesleykhuu committed:
added comments

added comments

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+Methods used to edit contacts. Used in actualContactEdit.php
```
@brief

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+		public static function GetByEmail($email){ 
+			include 'connection.php';
+			session_start();
+			global $user;
```
Globals always bad... have the individual files check to see whether they're logged in or not. Keep this class focused on just the contact

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+				header("Location: index.html");
+			}
+
+			global $userInfo;
```
Try not to use globals in classes

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+			$sql = "SELECT * FROM `contactlist` WHERE `email`='$email' AND `user`='$user'";
+			$result = $conn->query($sql);
+			$info = $result->fetch_assoc();
+
```
Remember we want these static methods to return a Contact object... so it should be something like new Contact in here and the return that contact.

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+		private $currentEmail;
+
+/**
+Instantiates the object by email
```
@brief

@param

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+/**
+Sets the value of a column you want to edit.
+$set can be firstname, lastname, or email
+$value is the new value you want to store
```
@param

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+			global $userInfo;
+			$userInfo[$set] = $value;
+			
+
```
whats the point of the connection.php here? KISS,

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+Gets the column from the database and returns it.
+*/
+		public function getValue($value){
+			global $userInfo;
```
no globals, use class variables... private userInfo declared at the top... and then pull from that

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,142 @@
+			$sql = "UPDATE `contactlist` SET `firstname`= '$firstname', `lastname`= '$lastname', `email`= '$email' WHERE `email` ='$currentEmail' AND `user` = '$user'";
+			$result = $conn->query($sql);
+
+
```
Probably want to return true or false on success or failure

ontraportgithub commented on crmlogin/actualContactEdit.php:
```diff
@@ -0,0 +1,48 @@
+	$check = 0;
+
+	if(!empty($_POST['firstname']) || !empty($_POST['lastname']) || !empty($_POST['email'])){
+		$obj = new Contact();
```
Here you're instantiating a new object.. but 2 lines later you're using that objects static method.

Static method should be used like this

$contact = Contact::GetById($contactId);

lesleykhuu committed:
changed Contact.php and comments

created the contact in GetById.
added @brief and @param where needed. got rid of GetByEmail

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,125 @@
+/**
+@brief Instantiates the object by contactId
+@param $id is the contactId of the contact that is being editted
+*/
```
Let's indent the comments so they're inline with the method indentation.

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,125 @@
+			$result = $conn->query($sql);
+			$info = $result->fetch_assoc();
+
+			$contact->userInfo['firstname'] = $info['firstname'];
```
Since the userInfo is private variables, this wouldn't work.

Use your helper methods, setValue and getValue

ontraportgithub commented on crmlogin/Contact.php:
```diff
@@ -0,0 +1,125 @@
+			$email = $this->userInfo['email'];
+			$contactId = $this->userInfo['contactId'];
+			
+			$sql = "UPDATE `contactlist` SET `firstname`= '$firstname', `lastname`= '$lastname', `email`= '$email' WHERE `contactId` = '$contactId'";
```
Remember to escape the fields.. even if they use setValue.. its untrusted inputs.

ontraportgithub commented on crmlogin/actualContactEdit.php:
```diff
@@ -0,0 +1,49 @@
+Takes in values from contactedit2.php and changes them in the database.
+
+*/
+	include 'Contact.php';
```
require instead of include. If the include fails... the rest of this sript is useless.

ontraportgithub replied:
Why indent?

lesleykhuu committed:
made the changes and used the methods in contactcreate

made the changes and used the methods in contactcreate
